src/rp2_common/boot_stage2/BUILD.bazel

```diff
@@ -0,0 +1,82 @@
+load("//bazel/toolchain:objcopy.bzl", "objcopy_to_bin")
+
+package(default_visibility = ["//visibility:private"])
+
+cc_library(
+    name = "config",
+    hdrs = [
+        "asminclude/boot2_helpers/exit_from_boot2.S",
+        "asminclude/boot2_helpers/read_flash_sreg.S",
+        "asminclude/boot2_helpers/wait_ssi_ready.S",
+        "boot2_at25sf128a.S",
+        "boot2_generic_03h.S",
+        "boot2_is25lp080.S",
+        "boot2_usb_blinky.S",
+        "boot2_w25q080.S",
+        "boot2_w25x10cl.S",
+        "include/boot_stage2/config.h",
+    ],
+    includes = ["include"],
+    visibility = ["//src/rp2_common/pico_standard_link:__pkg__"],
+)
+
+cc_binary(
+    name = "boot_stage2_elf",
+    srcs = ["compile_time_choice.S"],
+    copts = ["-fPIC"],
+    includes = [
+        "asminclude",
+        "include",
+    ],
+    linkopts = [
+        "-Wl,--no-gc-sections",
+        "-nostartfiles",
+        "-T$(location boot_stage2.ld)",
+    ],
+    deps = [
+        "boot_stage2.ld",
+        ":config",
+        "//src/common/pico_base:pico_base_interface",
+        "//src/common/pico_base:pico_platform",
+    ],
+)
+
+objcopy_to_bin(
+    name = "boot_stage2_bin",
+    src = ":boot_stage2_elf",
+    out = "boot_stage2.bin",
+)
+
+# WORKAROUND: Python rules always require a .py extension.
+genrule(
+    name = "copy_tool_to_py",
+    srcs = ["pad_checksum"],
+    outs = ["pad_checksum_tool.py"],
+    cmd = "cp $< $@",
+    cmd_bat = "copy $< $@",
+)
+
+py_binary(
+    name = "pad_checksum_tool",
+    srcs = ["pad_checksum_tool.py"],
+)
+
+_PAD_CHECKSUM_CMD = "$(location :pad_checksum_tool) -s 0xffffffff  $< $@"
+
+genrule(
```

Comment:
Consider using `run_binary` from Skylib, which doesn't rely on the shell (https://github.com/bazelbuild/bazel-skylib/blob/main/docs/run_binary_doc.md). From [the docs](https://bazel.build/reference/be/general#genrule):

> Note that genrule requires a shell to interpret the command argument. It is also easy to reference arbitrary programs available on the PATH, however this makes the command non-hermetic and may not be reproducible. If you only need to run a single tool, consider using [run_binary](https://github.com/bazelbuild/bazel-skylib/blob/main/docs/run_binary_doc.md) instead.